core/src/cubos/core/data/ser/json.cpp

```diff
@@ -0,0 +1,131 @@
+#include <cubos/core/data/ser/json.hpp>
+#include <cubos/core/log.hpp>
+
+using cubos::core::data::JSONSerializer;
+using cubos::core::reflection::Type;
+
+// Macro used to reduce code duplication for primitive type hooks.
+#define AUTO_HOOK(type, ...)                                                                                           \
+    this->hook<type>([this](const type& value) {                                                                       \
+        mJSON = value;                                                                                                 \
+        return true;                                                                                                   \
+    })
+
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using cubos::core::reflection::reflect;
+
+JSONSerializer::JSONSerializer()
+{
+    AUTO_HOOK(bool);
+    AUTO_HOOK(char);
+    AUTO_HOOK(int8_t);
+    AUTO_HOOK(int16_t);
+    AUTO_HOOK(int32_t);
+    AUTO_HOOK(int64_t);
+    AUTO_HOOK(uint8_t);
+    AUTO_HOOK(uint16_t);
+    AUTO_HOOK(uint32_t);
+    AUTO_HOOK(uint64_t);
```

Comment:
After rebasing on `origin/main` you'll have to change these hooks to the new primitive integer types.